src/types/allocator.rs

```diff
@@ -131,8 +166,16 @@ pub trait PoolableOps: Poolable + Sized {
 #[sealed]
 impl<T: Poolable> PoolableOps for T {
     fn alloc() -> Option<PoolRef<mem::MaybeUninit<Self>>> {
-        let result = unsafe { vault_alloc(T::Pool::vault(), mem::size_of::<T>() as u32)? };
-        (!result.is_null()).then(|| PoolRef(result.cast::<mem::MaybeUninit<Self>>()))
+        let mut result = AllocationResult::default();
+        let size = mem::size_of::<Self>();
+        unsafe {
+            let alloc = crate::fn_from_hash!(
+                Memory_Vault_Alloc,
+                unsafe extern "C" fn(*mut red::Memory::Vault, *mut AllocationResult, u32)
+            );
+            alloc(T::Pool::vault(), &mut result, size as _);
+        };
+        (!result.memory.is_null()).then(|| PoolRef(result.memory.cast::<mem::MaybeUninit<Self>>()))
```

Comment:
this looks identical to the previous code, what is the intended change in this PR?

Reply:
The change introduced to fix `RedHashMap` broke `ClassExport` for native classes.
I just restricted the fix to the allocator of the `RedHashMap`, reverting the rest.

Reply:
I might have missed something: I'm gonna wait that we merge the other PRs, run latest `master` with `audioware` and see if there's no issue. In case of error, I'll diff the 2 branches to provide further details.
Please hold on this PR in the meantime.